pwndbg/dbg/gdb.py

```diff
@@ -86,10 +87,12 @@ def __init__(self, inner: gdb.Frame):
         self.inner = inner
 
     @override
-    def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
-        from pwndbg.gdblib.scheduler import lock_scheduler
+    def evaluate_expression(
+        self, expression: str, lock_scheduler: bool = False
+    ) -> pwndbg.dbg_mod.Value:
+        from pwndbg.gdblib.scheduler import lock_scheduler as do_lock_scheduler
 
-        with lock_scheduler():
+        with do_lock_scheduler() if lock_scheduler else nullcontext():
```

Comment:
Some problem I see with it is that if u execute:

```
evaluate_expression(a, lock=True)
evaluate_expression(b, lock=True)
```
you probably want to lock the scheduler once and only once for all the execution.

Also, if the evaluate_expression would somehow trigger another `lock()` then maybe there could be some issue? idk.

Reply:
OTOH, they could then use lock scheduler outside? thats probably fine as it is now